rust/geoarrow-schema/src/type.rs

```diff
@@ -0,0 +1,1553 @@
+            Dimension::XYZM => {
+                vec![
+                    Field::new("xmin", DataType::Float64, false),
+                    Field::new("ymin", DataType::Float64, false),
+                    Field::new("zmin", DataType::Float64, false),
+                    Field::new("mmin", DataType::Float64, false),
+                    Field::new("xmax", DataType::Float64, false),
+                    Field::new("ymax", DataType::Float64, false),
+                    Field::new("zmax", DataType::Float64, false),
+                    Field::new("mmax", DataType::Float64, false),
+                ]
+            }
+        };
+        DataType::Struct(values_fields.into())
+    }
+}
+
+impl ExtensionType for BoxType {
+    const NAME: &'static str = "geoarrow.box";
+
+    type Metadata = Metadata;
+
+    fn metadata(&self) -> &Self::Metadata {
+        &self.metadata
+    }
+
+    fn serialize_metadata(&self) -> Option<String> {
+        self.metadata.serialize()
+    }
+
+    fn deserialize_metadata(metadata: Option<&str>) -> Result<Self::Metadata, ArrowError> {
+        Metadata::deserialize(metadata)
+    }
+
+    fn supports_data_type(&self, data_type: &DataType) -> Result<(), ArrowError> {
+        let dim = parse_box(data_type)?;
+        if dim != self.dim {
+            return Err(ArrowError::SchemaError(format!(
+                "Expected dimension {:?}, but got {:?}",
+                self.dim, dim
+            )));
+        }
+        Ok(())
+    }
+
+    fn try_new(data_type: &DataType, metadata: Self::Metadata) -> Result<Self, ArrowError> {
+        let dim = parse_box(data_type)?;
+        Ok(Self {
+            dim,
+            metadata: Arc::new(metadata),
+        })
+    }
+}
+
+fn parse_box(data_type: &DataType) -> Result<Dimension, ArrowError> {
+    match data_type {
+        DataType::Struct(struct_fields) => match struct_fields.len() {
+            4 => Ok(Dimension::XY),
+            6 => {
+                let names: HashSet<&str> =
+                    struct_fields.iter().map(|f| f.name().as_str()).collect();
+                if names.contains("mmin") && names.contains("mmax") {
+                    Ok(Dimension::XYM)
+                } else if names.contains("zmin") && names.contains("zmax") {
+                    Ok(Dimension::XYZ)
+                } else {
+                    Err(ArrowError::SchemaError(format!("unexpected either mmin and mmax or zmin and zmax for struct with 6 fields. Got names: {:?}", names)))
+                }
+            }
+            8 => Ok(Dimension::XYZM),
+            num_fields => Err(ArrowError::SchemaError(format!(
+                "unexpected number of struct fields: {}",
+                num_fields
+            ))),
+        },
+        dt => Err(ArrowError::SchemaError(format!(
+            "unexpected data type parsing box: {:?}",
+            dt
+        ))),
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct WkbType {
+    metadata: Arc<Metadata>,
+}
```

Comment:
@paleolimbot  Should there be a 1:1 correspondence from geoarrow type to `DataType`? 
For all other types, the `CoordType` and `Dimension` uniquely identify the corresponding `DataType` but this is not true for `WkbType` and `WktType` currently.

Should `WkbType` and `WktType` contain an attribute like `large` that defines whether it uses i32 or i64 offsets? I assume not because we want `supports_data_type` to allow either `DataType` presumably.